### PR TITLE
Add titleView to SelectMenu

### PR DESCRIPTION
**Pull request by notfelineit**
**This PR:**
- Fixes `onFilterChange` warning in the console
- Adds an optional tooltip `?` icon option to the SelectMenu header. (Needed for some Protocols UI work)

selectmenu

If this seems to specialized, and we should instead pass a render function for the header (or something else!) let me know :-)

### CODE REVIEW

**Comment by vadimdemedes** [comment]
I think this may be too small of a change to trigger such major addition as header render function.

What do you think about removing tooltip in favor of putting text below the header? Reason I'm suggesting this is to avoid 2 popovers at the same time (select menu and tooltip).

cleanshot 2019-01-18 at 19 53 21 2x

Either way, I'm good with this change 👍

File: src/select-menu/src/SelectMenuContent.js [hunk 1]
```diff
@@ -45,9 +49,11 @@ export default class SelectMenuContent extends PureComponent {
       height,
       options,
       hasTitle,
+      hasTooltip,
       hasFilter,
       close,
       title,
+      tooltipContent,
```
Comment (vadimdemedes) [inline src/select-menu/src/SelectMenuContent.js]: Could you rename this prop to `tooltip`, so that it matches `hasTooltip` prop naming?

**Comment by notfelineit** [comment]
@vadimdemedes That's a good idea! Will it work with really long helper text?

**Comment by notfelineit** [comment]
@mshwery I agree 🤔 Before the tooltip, I had a subtitle (like Vadim's suggestion) with the following: `To create a new tag type a key:value format into the search field`. But that wrapped and made the title wonky, hence the tooltip. Do you guys have any suggestions on how else to surface this to users within the SelectMenu? Maybe as a detailPane?

File: src/select-menu/src/SelectMenuContent.js [hunk 2]
```diff
@@ -47,15 +51,16 @@ export default class SelectMenuContent extends PureComponent {
       hasTitle,
       hasFilter,
       close,
-      title,
       listProps,
+      titleView,
       detailView,
       emptyView,
+      headerHeight,
```
Comment (mshwery) [inline src/select-menu/src/SelectMenuContent.js]: Should we provide a default value?

Comment (notfelineit) [reply src/select-menu/src/SelectMenuContent.js]: Absorootintootinly

File: src/select-menu/src/SelectMenuContent.js [hunk 3]
```diff
@@ -68,26 +74,7 @@ export default class SelectMenuContent extends PureComponent {
...
-              </Pane>
-              <IconButton
-                icon="cross"
-                appearance="minimal"
-                height={24}
-                onClick={close}
-              />
-            </Pane>
-          )}
+          {hasTitle && titleView}
```
Comment (mshwery) [inline src/select-menu/src/SelectMenuContent.js]: Hm, I think we should provide the below (removed) jsx as the _default prop_ value for `titleView`. That way consumers of this component can continue using it as is.

The way you've moved the code around makes this a breaking change, technically.

File: src/select-menu/src/SelectMenu.js [hunk 4]
```diff
@@ -128,6 +138,40 @@ export default class SelectMenu extends PureComponent {
     return {}
   }
 
+  getTitleView = (close, title, headerHeight, titleView) => {
```
Comment (mshwery) [inline src/select-menu/src/SelectMenu.js]: I think you could get rid of this code in favor pushing it into the component that actually uses `titleView` instead.

File: src/select-menu/src/SelectMenuContent.js [hunk 5]
```diff
@@ -36,26 +42,48 @@ export default class SelectMenuContent extends PureComponent {
   static defaultProps = {
     options: [],
     hasTitle: true,
-    hasFilter: true
+    hasFilter: true,
+    titleView: ({ close, title, headerHeight }) => (
```
Comment (mshwery) [inline src/select-menu/src/SelectMenuContent.js]: Nit: I would pull this out into a separate component so the defaultProps are easier to scan.

But this is exactly what I was thinking of!

Comment (notfelineit) [reply src/select-menu/src/SelectMenuContent.js]: Whoo! ok, I'll do that.

**Pull request merged**